rocketc/rocketc.py

```diff
@@ -468,3 +487,92 @@ def _set_topic(self, group_id, topic):
         response = self._request_rocket_chat(method, url_path, data)
 
         LOG.info("Method Set Topic: %s with this data: %s", response, data)
+
+    def _teams_is_enabled(self):
+        """
+        This method verifies if teams are available
+        """
+        from openedx_dependencies import modulestore  # pylint: disable=relative-import
+        try:
+            course_id = self.runtime.course_id  # pylint: disable=no-member
+        except AttributeError:
+            return False
+
+        course = modulestore().get_course(course_id, depth=0)
+        teams_configuration = course.teams_configuration
+        LOG.info("Team is enabled result: %s", teams_configuration)
+        if "topics" in teams_configuration and teams_configuration["topics"]:
+            return True
+
+        return False
+
+    def _get_team(self, username, course_id):
+        """
+        This method gets the user's team
+        """
+        try:
+            user = self.xblock_settings["username"]
+            password = self.xblock_settings["password"]
+            client_id = self.xblock_settings["client_id"]
+            client_secret = self.xblock_settings["client_secret"]
+        except KeyError:
+            raise
+
+        server_url = settings.LMS_ROOT_URL
+
+        api = ApiTeams(user, password, client_id, client_secret, server_url)
+        team = api.get_user_team(course_id, username)
+        LOG.info("Get Team response: %s", team)
+        if team:
+            return team[0]
+        return None
+
+    def _add_to_team_group(self, user_id, username, course_id):
+        """
+        Add the user to team's group in rocketChat
+        """
+        team = self._get_team(username, course_id)
+
+        if team is None:
+            return False
+
+        group_name = "-".join(["Team", team["topic_id"], team["name"]])
+        group_info = self._search_rocket_chat_group(group_name)
+        self.team_channel = group_name
+
+        if group_info["success"]:
```

Comment:
I'm thinking if we should encapsulate the logic referent to validate if a group exists inside the methods _create_group and add_to_group. The reason is to prevent duplicating code every time we need to add someone to a course validating if the course exists or not. This can be included in the refactor

Reply:
The decision to create or add a user depends on the search_rocket_chat's result

Reply:
We can have a method like _get_or_create_group which validates first if the group exists or not